integration-cli/docker_cli_rename_test.go

```diff
@@ -38,6 +38,27 @@ func (s *DockerSuite) TestRenameRunningContainer(c *check.C) {
 	c.Assert(name, checker.Equals, "/"+newName, check.Commentf("Failed to rename container %s", name))
 }
 
+func (s *DockerSuite) TestRenameRunningContainerAndReuse(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	out, _ := dockerCmd(c, "run", "--name", "first_name", "-d", "busybox", "top")
+	c.Assert(waitRun("first_name"), check.IsNil)
+
+	newName := "new_name" + stringid.GenerateNonCryptoID()
+	cleanedContainerID := strings.TrimSpace(out)
+	dockerCmd(c, "rename", "first_name", newName)
+
+	name, err := inspectField(cleanedContainerID, "Name")
+	c.Assert(err, checker.IsNil, check.Commentf("Failed to rename container %s", name))
+	c.Assert(name, checker.Equals, "/"+newName, check.Commentf("Failed to rename container %s", name))
```

Comment:
@cpuguy83 Since the name is from the inspect it will be useful for debugging when the test fails because of expecting and actual names not matching.. Will address the other two comments.